src/fault_injector/fault_types/platform_blocked_fault.py

```diff
@@ -1,11 +1,10 @@
-from src.component import Component
 from src.fault_injector.fault_types.fault import Fault
 
 
 class PlatformBlockedFault(Fault):
     """A fault that blocks a platform"""
 
-    def inject_fault(self, component: Component):
+    def inject_fault(self):
         """inject PlatformBlockedFault into the given component
 
         :param component: The component the fault should be injected into
```

Comment:
It will be replaced by more specific attributes.  For example, logger, wrapper or interlocking. In addition, the number of components needed is not always the same.